processing/src/main/java/io/druid/segment/IndexSpec.java

```diff
@@ -91,7 +94,8 @@ public IndexSpec()
   public IndexSpec(
       @JsonProperty("bitmap") BitmapSerdeFactory bitmapSerdeFactory,
       @JsonProperty("dimensionCompression") String dimensionCompression,
-      @JsonProperty("metricCompression") String metricCompression
+      @JsonProperty("metricCompression") String metricCompression,
+      @JsonProperty("metricLongEncoding") String metricLongEncoding
```

Comment:
This is more like a `longEncoding`, assuming it will apply to long dimensions and `__time` too.